plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -382,6 +390,8 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
          auto before = _unapplied_transactions.size();
          _unapplied_transactions.clear_applied( bsp );
          _subjective_billing.on_block( _log, bsp, fc::time_point::now() );
+         //_snapshot_scheduler.on_block(bsp->block_num);
```

Comment:
Remove. Although, I wonder why you are calling snapshot_scheduler on blocks start instead of on accepted block.

Reply:
It would be nice to move all the existing snapshot generation code out of producer_plugin into the snapshot_scheduler or related classes.

Reply:
on_start_block was the only hook that allowed execution of existing implementation of create_snapshot without massive chain of hard to trace exceptions. on_start_block hook executes create+snapshot perfectly with no issues.